### PR TITLE
Allow all users to login using e-mail

### PR DESCRIPTION
**Pull request by mrk-j**
This PR adds `allowAllDomains()` method.

See #6.

### CODE REVIEW

File: src/Filament/Pages/Login.php [hunk 1]
```diff
@@ -24,42 +24,50 @@ public function authenticate(): ?LoginResponse
...
+            $user = null;
 
-            if ($email->endsWith($plugin->getDomains())) {
+            if ($plugin->areAllDomainsAllowed()) {
+                $user = ($plugin->getModelClass())::query()
+                    ->where($plugin->getModelColumn(), '=', $email)
+                    ->first();
+            }
+
+            if (! $user && $email->endsWith($plugin->getDomains())) {
```
Comment (m-bosch) [inline src/Filament/Pages/Login.php:35]: it is correct that when `$plugin->areAllDomainsAllowed()` isn't set (`false`) the user is null. 

When `$plugin->areAllDomainsAllowed()` isset, and the e-mailaddress isn't found in the part of line 30-32, should it be logic to still check if there is ANY user with the domain? 

E.G. when `$plugin->areAllDomainsAllowed()` isset, and domain is set to `gmail.com`. some user like `johndoe@gmail.com` cant be found, but he will pick the first user with domain `gmail.com`. This way that first picked up user will receive an unknown / unwanted login e-mail?

long story short, shouldn't we review what options can be stacked or not.
